src/env.cc

```diff
@@ -383,6 +383,10 @@ Environment::Environment(IsolateData* isolate_data,
   inspector_host_port_ = std::make_shared<ExclusiveAccess<HostPort>>(
       options_->debug_options().host_port);
 
+  if (!options_->global_search_paths) {
+    flags_ |= EnvironmentFlags::kNoGlobalSearchPaths;
+  }
```

Comment:
nit: I would put this into the implementation of `no_global_search_paths()`, i.e. return `(flags_ & EnvironmentFlags::kNoGlobalSearchPaths) || !options()->global_search_paths` there. That way, it is also honored if at some point there will be a way to change options at runtime.